### ISSUE TITLE
`clippy::zombie_processes` incorrectly detects early return in a spawned thread closure

### ISSUE THREAD

**Issue by anatols** [rust-lang/rust-clippy#16241]
### Summary

The `clippy::zombie_processes` detects an early return in a closure that's spawned in a background thread and complains about a missing wait for a child process.

I believe this might be related to the fix in https://github.com/rust-lang/rust-clippy/issues/13459.

### Lint Name

zombie_processes

### Reproducer

I tried this code:

```rust
fn main() {
    let mut child = std::process::Command::new("ls")
        .stdout(std::process::Stdio::piped())
        .spawn()
        .unwrap();

    let child_stdout = child.stdout.take().unwrap();
    std::thread::spawn(move || -> Result<(), std::io::Error> {
        use std::io::BufRead;
        for line in std::io::BufReader::new(child_stdout).lines() {
            let line = line?;
            println!("{line}");
        }
        Ok(())
    });

    child.wait().unwrap();
}
```

I saw this happen:

```
warning: spawned process is not `wait()`ed on in all code paths
  --> src/main.rs:2:21
   |
2  |       let mut child = std::process::Command::new("ls")
   |  _____________________^
3  | |         .stdout(std::process::Stdio::piped())
4  | |         .spawn()
5  | |         .unwrap();
   | |_________________^
   |
note: no `wait()` call exists on the code path to this early return
  --> src/main.rs:11:28
   |
11 |             let line = line?;
   |                            ^
note: `wait()` call exists, but it is unreachable due to the early return
  --> src/main.rs:17:5
   |
17 |     child.wait().unwrap();
   |     ^^^^^
   = help: consider calling `.wait()` in all code paths
   = note: not doing so might leave behind zombie processes
   = note: see https://doc.rust-lang.org/stable/std/process/struct.Child.html#warning
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#zombie_processes
   = note: `#[warn(clippy::zombie_processes)]` on by default

```

I expected to see this happen:
No warning.

### Version

```text
rustc 1.87.0 (17067e9ac 2025-05-09)
binary: rustc
commit-hash: 17067e9ac6d7ecb70e50f92c1944e545188d2359
commit-date: 2025-05-09
host: x86_64-unknown-linux-gnu
release: 1.87.0
LLVM version: 20.1.1
```

**Comment by samueltardieu** [comment]
~~Pinging @y21 in case they are interested in grabbing this.~~

Nevermind, you fixed this already, which is why I didn't understand how the existing code could fail!

**Comment by samueltardieu** [comment]
It has been fixed since Clippy 1.89 (or maybe even 1.88).
